packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -38,12 +41,25 @@ export const KvCodeEditor = ({
 		editorRef.current.updateOptions(editorOptions);
 	}, [editorRef.current, readOnlyRef.current, editorOptions]);
 
+	const loadFont = useCallback(async () => {
+		await document.fonts.load(getFontName(initialOptions));
+		setFontLoaded(true);
+	}, []);
+
 	useEffect(() => {
 		setupEditor();
 		updateOptions();
 	}, [setupEditor, updateOptions]);
 
-	return <Editor language={language} theme={theme} options={DEFAULT_MONACO_OPTIONS} value={code} loading={loadingComponent} onMount={onEditorMount} onChange={onTextChange} />;
+	useEffect(() => {
+		loadFont();
+	}, []);
+
+	return isFontLoaded ? (
+		<Editor language={language} theme={theme} options={initialOptions} value={code} loading={loadingComponent} onMount={onEditorMount} onChange={onTextChange} />
+	) : (
+		<>{loadingComponent}</>
+	);
```

Comment:
Let's keep jsx simple

```suggestion
    if (isFontLoading) {
        return loadingComponent;
    }
    
	return (
	    <Editor 
	        language={language}
	        theme={theme}
	        options={initialOptions}
	        value={code}
	        loading={loadingComponent}
	        onMount={onEditorMount}
	        onChange={onTextChange} 
	    />
```